cmd/krel/cmd/anago/push_git_objects.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package anago
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"k8s.io/release/pkg/git"
+	"k8s.io/release/pkg/release"
+	"k8s.io/release/pkg/util"
+)
+
+// pushGitObjectsCmd is the krel push-git-objects subcommand
+var pushGitObjectsCmd = &cobra.Command{
+	Use:   "push-git-objects",
+	Short: "Push branches and tags to the remote github repository",
+	Long: `krel push-git-objects
+
+NOTE: This subcommand is a temporary commando to be invoked from anago.
+
+The purpose of krel push-git-objects is to push branches and tags to the
+git remote repository of kubernetes. 
+`,
+	SilenceUsage:  true,
+	SilenceErrors: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		err := runPushGitObjects(pushGitObjectsOpts)
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+}
+
+type pushGitObjectsOptions struct {
+	nomock        bool
+	sayYes        bool
+	maxRetries    int
+	releaseBranch string
+	parentBranch  string
+	repoPath      string
+	tags          []string
+}
+
+var pushGitObjectsOpts = &pushGitObjectsOptions{}
+
+func init() {
+	pushGitObjectsCmd.PersistentFlags().StringVar(
+		&pushGitObjectsOpts.releaseBranch,
+		"release-branch",
+		"",
+		"branch name to push",
+	)
+
+	pushGitObjectsCmd.PersistentFlags().StringVar(
+		&pushGitObjectsOpts.parentBranch,
+		"parent-branch",
+		"",
+		fmt.Sprintf("parent branch, if different from %s", git.DefaultBranch),
+	)
+
+	pushGitObjectsCmd.PersistentFlags().BoolVarP(
+		&pushGitObjectsOpts.sayYes,
+		"yes",
+		"y",
+		false,
+		"say yes to prompts without asking the user",
+	)
```

Comment:
I'd prefer removing this and defaulting to `yes`. `anago` is not executed manually by a human usually.

Reply:
OK, I removed it. If we still want to prompt the user we'll have to keep this prompt in bash:
https://github.com/kubernetes/release/blob/b067ca634072851e41baaf71cb27a543bce3dc95/anago#L766-L769